app/lib/system/redis_pool.rb

```diff
@@ -6,11 +6,11 @@ module System
   # RedisPool a simple wrapper around Redis with connection pooling
   class RedisPool
 
-    def initialize(config={})
-      config = config.dup
-      pool_config = config.extract!(:pool_size, :pool_timeout)
-      @pool = ConnectionPool.new(size: pool_config[:pool_size] || 5, timeout: pool_config[:pool_timeout] || 5 ) do
-        Redis.new_with_namespace(config)
+    def initialize(config = {})
+      cfg = config.to_h
+      pool_config = cfg.extract!(:size, :pool_timeout)
+      @pool = ConnectionPool.new(size: pool_config[:size] || 5, timeout: pool_config[:pool_timeout] || 5 ) do
+        Redis.new_with_namespace(cfg)
```

Comment:
The old `:pool_size` parameter is rejected by the Redis client and we need to use `:size` now. This reflects the change. Also renamed `config` to `cfg` for better understanding.

Reply:
I don't see an improvement calling it `cfg` but whatever :)